napari/components/layerlist.py

```diff
@@ -108,70 +117,60 @@ def unselect_all(self, ignore=None):
         ignore : Layer | None
             Layer that should not be unselected if specified.
         """
-        for layer in self:
-            if layer.selected and layer != ignore:
-                layer.selected = False
+        self.selection.intersection_update({ignore} if ignore else {})
 
     def select_all(self):
         """Selects all layers."""
-        for layer in self:
-            if not layer.selected:
-                layer.selected = True
+        self.selection.update(self)
 
     def remove_selected(self):
         """Removes selected items from list."""
-        to_delete = []
-        for i in range(len(self)):
-            if self[i].selected:
-                to_delete.append(i)
-        to_delete.reverse()
-        for i in to_delete:
-            self.pop(i)
-        if len(to_delete) > 0:
-            first_to_delete = to_delete[-1]
-            if first_to_delete == 0 and len(self) > 0:
-                self[0].selected = True
-            elif first_to_delete > 0:
-                self[first_to_delete - 1].selected = True
+        for i in reversed(self):
+            if i in self.selection:
+                self.remove(i)
+
+        # FIXME
```

Comment:
As discussed on other PR/ zulip - I don't have strong feelings about this, can leave none selected or can just highlight the top most layer above the previous selection

Reply:
thanks. I went with the simple behavior: `remove_selected` now just removes the currently selected layers.  Many of these methods seem to be there mostly for the qt_layerlist to use, and may be removed altogether when we move to the new qt layerlist

Reply:
> Many of these methods seem to be there mostly for the qt_layerlist to use, and may be removed altogether when we move to the new qt layerlist

Yes that's correct, happy to deprecate/ trim then as we need